apps/web/app/api/cancel/route.ts

```diff
@@ -5,18 +5,26 @@ import type { NextRequest } from "next/server";
 
 import { getServerSession } from "@calcom/features/auth/lib/getServerSession";
 import handleCancelBooking from "@calcom/features/bookings/lib/handleCancelBooking";
+import { checkRateLimitAndThrowError } from "@calcom/lib/checkRateLimitAndThrowError";
+import getIP from "@calcom/lib/getIP";
 import { bookingCancelInput } from "@calcom/prisma/zod-utils";
 
 import { buildLegacyRequest } from "@lib/buildLegacyCtx";
 
 async function handler(req: NextRequest) {
+  const ip = getIP(req);
+  await checkRateLimitAndThrowError({
+    rateLimitingType: "core",
+    identifier: `booking.cancel.${ip}`,
+  });
```

Comment:
Missing the _ip_ existence check here. 

I think it might help to have a reusable fn like, 

```
checkIpRateLimitAndThrowError({
  ip: ip,
  identifierPrefix: 'booking.cancel.',
})
```